### PR TITLE
ENH: EDAX I/O plugin

### LINKED ISSUE #756: Importing of EDAX EDS data
Having recently acquired an EDAX EDS/EBSD system, it has made me realize that there is no simple way to read the EDAX .spd (maps), .spc (spectra), and .ipr (image) files produced by EDAX's TEAM software in Hyperspy. 

I contacted EDAX applications and they provided me with the binary specifications for each of these files, so it shouldn't be too tricky to be able to implement at least read (and maybe even write) functionality for these formats. 

Apparently I cannot upload these documents without write permission to the repo, but they are available here: https://umd.box.com/s/nf4eqti4g07b76hh9zud25h6y1b6mmht

I am going to begin working on this in the next few days and will create a PR when I have some progress, but I wanted to put this issue here so we had a record of it. Any suggestions or tips would be be appreciated!

### PR DESCRIPTION
**Pull request by jat255**
Eventually will Resolve #756.

Submitting the PR early so people can comment along the way and prod me in the right direction....

TODO:
- [x] Figure out appropriate offsets and dtypes for reading binary formats
- [x] Allow for custom .ipr and .spc filenames
- [x] User guide documentation
- [x] Actual import of data
- [x] `file_reader` function
- [ ] Unit tests

### CODE REVIEW

**Comment by francisco-dlp** [comment]
Looks really good. I would add unittests to your TODO list.

**Comment by jat255** [comment]
Okay, sorry for the delay in getting back to this, but it seems to be working on my test data and I think is ready for some review. I rebased onto the most recent master, because there were a number of conflicts (my apologies if someone had forked this branch before).

Paging @francisco-dlp, @vidartf, @to266, @AndrewHerzing, and whoever else might be interested. Like I said, this is really my first ever attempt at file reading, so things might not be done in the best way. In my tests, it is pretty efficient and works okay on the few different test files I have. 

I know that it still needs test cases. I'll see if I can't figure out how to add those soon.

File: hyperspy/io_plugins/edax.py [hunk 1]
```diff
@@ -0,0 +1,935 @@
...
+                   68: 'Er', 69: 'Tm', 70: 'Yb', 71: 'Lu', 72: 'Hf', 73: 'Ta',
+                   74: 'W', 75: 'Re', 76: 'Os', 77: 'Ir', 78: 'Pt', 79: 'Au',
+                   80: 'Hg', 81: 'Tl', 82: 'Pb', 83: 'Bi', 84: 'Po', 85: 'At',
+                   86: 'Rn', 87: 'Fr', 88: 'Ra', 89: 'Ac', 90: 'Th', 91: 'Pa',
+                   92: 'U', 93: 'Np', 94: 'Pu', 95: 'Am', 96: 'Cm', 97: 'Bk',
+                   98: 'Cf', 99: 'Es', 100: 'Fm', 101: 'Md', 102: 'No',
+                   103: 'Lr', 104: 'Rf', 105: 'Db', 106: 'Sg', 107: 'Bh',
+                   108: 'Hs', 109: 'Mt', 110: 'Ds', 111: 'Rg', 112: 'Cp',
+                   113: 'Uut', 114: 'Uuq', 115: 'Uup', 116: 'Uuh', 117: 'Uus',
+                   118: 'Uuo'}
```
Comment (vidartf) [inline hyperspy/io_plugins/edax.py]: Could be replaced by:
```python
from hyperspy.misc.elements import elements_db
atomic_num_dict = dict(((p.General_properties.Z, e) for (e, p) in elements_db)
```
But completely unimportant :)
(also the database stops at U, but that can be extended if need be)

File: hyperspy/io_plugins/edax.py [hunk 2]
```diff
@@ -0,0 +1,935 @@
...
+
+# Plugin characteristics
+# ----------------------
+format_name = 'EDAX TEAM'
+description = 'Reader for EDS maps and spectra saved by the EDAX TEAM' \
+              'software: \n' \
+              'An SPD file contains map data. The spectral information is ' \
+              '\nheld in an SPC file with the same name, while the spatial ' \
+              '\ncalibration is held in a related IPR file. If an SPD file ' \
+              '\nis loaded, the result will be a Hyperspy EDSSpectrum map, ' \
```
Comment (tjof2) [inline hyperspy/io_plugins/edax.py]: "HyperSpy" ;-)

**Comment by francisco-dlp** [comment]
This branch was automatically closed when we renamed `master` `RELEASE_next_major` and it cannot be reopened. Could you resubmitted it to RELEASE_next_minor instead?

**Pull request closed without merging**